gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,301 @@
+    private static final String IDENTIFIER_SYSTEM = "https://PSSAdaptor/";
+    private static final String VACCINE_PROCEDURE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect"
+        + "-VaccinationProcedure-1";
+    private static final String END_DATE_PREFIX = "End Date: ";
+    private static final String VACCINATION_CODING_EXTENSION_URL = "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding"
+        + "-sctdescid";
+
+    private CodeableConceptMapper codeableConceptMapper;
+
+    public List<Immunization> mapToImmunization(RCMRMT030101UK04EhrExtract ehrExtract, Patient patientResource,
+        List<Encounter> encounterList) {
+        List<Immunization> mappedImmunizationResources = new ArrayList<>();
+        var ehrCompositionList = EhrResourceExtractorUtil.extractValidImmunizationEhrCompositions(ehrExtract);
+
+        ehrCompositionList.forEach(ehrComposition -> {
+            var immunizationObservationStatements = getImmunizationObservationStatements(ehrComposition);
+
+            immunizationObservationStatements
+                .forEach(observationStatement -> {
+                    var mappedImmunization = mapImmunization(ehrComposition, observationStatement, patientResource,
+                        encounterList);
+                    mappedImmunizationResources.add(mappedImmunization);
+                });
+        });
+
+        return mappedImmunizationResources;
+    }
+
+    private Immunization mapImmunization(RCMRMT030101UK04EhrComposition ehrComposition,
+        RCMRMT030101UK04ObservationStatement observationStatement,
+        Patient patientResource, List<Encounter> encounterList) {
+
+        ImmunizationMapperParameters immunizationMapperParameters = new ImmunizationMapperParameters();
+
+        var id = observationStatement.getId().getRoot();
+        var identifier = getIdentifier(id);
+        var note = buildNote(observationStatement);
+        var date = getObservationDate(observationStatement);
+
+        Extension recordedTimeExtension = getRecordedTimeExtension(ehrComposition);
+        Extension vaccineExtension = createVaccineProcedureExtension(observationStatement);
+
+        var practitioner = ParticipantReferenceUtil.getParticipantReference(observationStatement.getParticipant(), ehrComposition);
+        var encounter = getEncounterReference(encounterList, ehrComposition.getId());
+
+        immunizationMapperParameters.setIdParam(id);
+        immunizationMapperParameters.setIdentifierParam(identifier);
+        immunizationMapperParameters.setNoteParam(note);
+        immunizationMapperParameters.setDateParam(date);
+        immunizationMapperParameters.setRecordedTimeExtensionParam(recordedTimeExtension);
+        immunizationMapperParameters.setVaccineExtensionParam(vaccineExtension);
+        immunizationMapperParameters.setPatientParam(new Reference(patientResource.getIdElement()));
```

Comment:
Sorry Alan, probably explained badly from me, all you need to do here is change this parameter to accept a Patient instead of a reference and pass the patientResource. That should eeal with everything itself (so no need to create any Reference object)